src/plugins/paste-image/index.ts

```diff
@@ -0,0 +1,42 @@
+import { PluginController } from "../PluginController";
+
+export default class PasteImage extends PluginController {
+  static id = "paste-image" as const;
+  static enabledByDefault = true;
+
+  pasteHandler = this._pasteHandler.bind(this);
+
+  afterEnable() {
+    document.addEventListener("paste", this.pasteHandler);
+  }
+
+  afterDisable() {
+    document.removeEventListener("paste", this.pasteHandler);
+  }
+
+  _pasteHandler(e: ClipboardEvent) {
+    if (!this.calc.selectedExpressionId) {
```

Comment:
place this `if` statement inside `if (clipboardFiles?.length){ ... }` to prevent any interaction when there are no images to paste

Reply:
I didn't realize that. Thanks for your advice!